packages/react-scripts/config/paths.js

```diff
@@ -117,3 +118,40 @@ if (
   };
 }
 // @remove-on-eject-end
+
+module.exports.srcPaths = [module.exports.appSrc];
+
+// recursively find package dirs
+// -- omit things in node_modules, nested packages, and current app package
+const findPkgs = root =>
+  fs
+    .readdirSync(root)
+    .filter(f => ['node_modules', '.git'].indexOf(f) === -1)
+    .map(f => path.join(root, f))
+    .filter(f => fs.realpathSync(f) !== appDirectory)
+    .filter(f => fs.statSync(f).isDirectory())
+    .reduce(
+      (pkgs, dir) =>
+        fs.existsSync(path.join(dir, 'package.json'))
+          ? pkgs.concat([dir])
+          : pkgs.concat(findPkgs(dir)),
+      []
+    );
+
+if (checkForMonorepo) {
+  // if app is in a monorepo (lerna or yarn workspace), allow any module inside
+  // the monorepo to be linted, transpiled, and tested as if it came from app's
+  // src.
+  const monoPkgPath = findPkg.sync(resolveApp('..'));
+  if (monoPkgPath) {
+    if (
+      require(monoPkgPath).workspaces ||
+      fs.existsSync(path.resolve(path.dirname(monoPkgPath), 'lerna.json'))
```

Comment:
Do you see any downsides to supporting Lerna? Does it make anything more complex, or is it just this single condition?

I’m asking because I think it‘s better to support one tool well than two tools poorly. My experience with Lerna alone has been mixed, and I wonder if we should tell users to just use Yarn Workspaces (and then can still add Lerna on top if they want to). But I can see how this wouldn’t be great for people who prefer npm.

Ultimately I don’t have a strong opinion here. I just want to make sure that our Lerna support will work just as well as YW. If you’re sure that’s the case (at least as far as we can do) then I’m cool with it.

Reply:
Yeah, lerna+npm basically does the same thing as YW, I didn't add anything extra to support it other than this one line.  And it should work with others, too, since they all rely on standard node module resolution to do their magic.  I can add an e2e test using lerna, probably should. (Uggh, I knew I should've called it e2e-monorepos).  I don't have a preference, tho.

Reply:
Anything I can do to make supporting Lerna more comfortable, please let me know. As it is now, we defer to YW during bootstrap, if configured to use yarn. It is effectively the same, although there are notable differences in the symlinks that result (as long as you're using the node resolution algorithm, you'll be fine).